### ISSUE TITLE
Mouse scrolling on macOS trackpad is far too sensitive

### ISSUE THREAD

**Issue by roachcord3** [hydrusnetwork/hydrus#710]
## Environment

Hydrus version: 415 from executable
Platform: macOS
OS specifics: macOS Catalina 10.15.7

## What happens

When scrolling with the macOS trackpad, any action bound to a mouse scroll will be triggered many times in quick succession for even small finger movements. For example, if scrolling is bound to next/previous in media navigation, scrolling will make you switch between several images at once.

I do not know if this will be a problem for other operating systems when using a trackpad, I only have one for my mac.

## Steps to reproduce

1. Bind scroll up/down to any action that can be easily repeated, such as `media navigation: next/previous`. (This is the default, I think.) I discourage anyone with photosensitivity from attempting this with media navigation actions though.
2. Scroll with the trackpad at anything more than a snail's pace.
3. Watch in horror as the images change way too rapidly (several times a second)

## Desired outcome

Ideally, one movement of the fingers would equal one scroll input. As in, I should have to lift my fingers off the trackpad, put them back down, and repeat the motion to scroll again.

## Error message / Log file / Screenshots

Not sure if logs will help at all here.

**Comment by bbappserver** [comment]
It doesn't seem to be high sensitivity so much has hydrus only cares if a scroll event happened and in which direction, ignoring velocity.  This works fine for clicky scroll wheels, but poorly for anything that scrolls smoothly.

Arguably it is working as intended in that the shortcut manager only has a field for mouse events and the event you emitted is certainly wheel up or down.  The problem is scroll is a special case of gesture for a shortcut event, where what you actually want is scroll and velocity on axis > threshold.

For example Tinder Gestures
```
Keep := scroll.velocity.x > 0.1 && scroll.inputState == InputRelease)
Delete := scroll.velocity.x < 0.1 && scroll.inputState == InputRelease)
```

**Comment by hydrusnetwork** [comment]
Thank you for this report. I'll see if I can add a new scroll event or similar to deal with gesture-based input better. I've had a similar problem with file position dragging on touchscreens.

**Comment by hydrusnetwork** [comment]
Thanks again. For tomorrow's release, I have written a routine that tries to recognise trackpad scrolling and then adapt the rate of scroll event generation according to your current velocity. I fudged a number here and am not sure how well this will work IRL, so please let me know.

This is only a change for my shortcut system, which generally governs more 'discrete' scrolling changes like in the media viewer previous/next or zoom bumping. More native stuff like scrolling through a tall thumbnail grid should generally be all handled by Qt, so I hope that already trackpad-scrolls well for you.

**Comment by roachcord3** [comment]
Thanks, I've been testing this a bit and here are my findings:

First off, before this, I was using the scroll to zoom in images. Since the behavior before had been very rapid I set my zoom levels to this:

`0.5,0.6,0.7,0.8,0.9,1.0,1.03,1.06,1.09,1.12,1.15,1.18,1.21,1.24,1.27,1.3,1.33,1.36,1.39,1.42,1.45,1.48,1.51,1.54,1.57,1.6,1.63,1.66,1.69,1.72,1.75,1.78,1.81,1.84,1.87,1.9,1.93,1.96,1.99,2.02,2.05,2.08,2.11,2.14,2.17,2.2,2.23,2.26,2.29,2.32,2.35,2.38,2.41,2.44,2.47,2.5,2.53,2.56,2.59,2.62,2.65,2.68,2.71,2.74,2.77,2.8,2.83,2.86,2.89,2.92,2.95,2.98,3.01,3.04,3.07,3.1,3.13,3.16,3.19,3.22,3.25,3.28,3.31,3.34,3.37,3.4,3.43,3.46,3.49,3.52,3.55,3.58,3.61,3.64,3.67,3.7,3.73,3.76,3.79,3.82,3.85,3.88,3.91,3.94,3.97,4.0`

The net effect was zooming was slow and to get to 4x zoom I had to do a full scroll motion at least twice, IME.

Now, the current behavior is that a scroll gesture seems to "cut off" sooner so it takes several full scrolls to reach the 4x zoom. Like, it took me around a dozen scrolls to do that.

It *is* definitely aware of velocity and I do like that so I might just adjust the zoom intervals, so this is not a complaint or anything, just letting you know what the effect is like so you know whether or not it fits what you want your software to act like.
